src/middlewared/middlewared/apps/webshell_app.py

```diff
@@ -200,7 +197,7 @@ def abort(self):
         asyncio.run_coroutine_threadsafe(self.ws.close(), self.loop)
 
         with contextlib.suppress(ProcessLookupError):
-            os.kill(self.shell_pid, signal.SIGTERM)
+            os.killpg(os.getpgid(self.shell_pid), signal.SIGTERM)
```

Comment:
And if they don't respect `SIGTERM`?

Reply:
Then someone is doing something via the webshell we never designed it for and we've probably got other issues. However, it's a valid point so I extended `terminate_pid` and call that instead. Thanks!